features/admin.connections.v1/pages/authenticator-edit.tsx

```diff
@@ -0,0 +1,371 @@
+     */
+    const getCustomLocalAuthTemplateId = (connector: CustomAuthConnectionInterface): string => {
+        const tags: string[] = (connector as CustomAuthConnectionInterface)?.tags || [];
+
+        const has2FA: boolean = tags.some((tag: string) => tag.toUpperCase() === AuthenticatorLabels.SECOND_FACTOR);
+
+        return has2FA
+            ? ConnectionTemplateIds.TWO_FACTOR_CUSTOM_AUTHENTICATION
+            : ConnectionTemplateIds.INTERNAL_CUSTOM_AUTHENTICATION;
+    };
+
+    /**
+     * Retrieves custom local authenticator details from the API.
+     *
+     * @param id - authenticator id.
+     */
+    const getCustomLocalAuthenticator = (id: string): void => {
+        setConnectorDetailFetchRequestLoading(true);
+
+        getLocalAuthenticator(id)
+            .then((response: CustomAuthConnectionInterface) => {
+                setConnector(response as CustomAuthConnectionInterface);
+            })
+            .catch((error: IdentityAppsApiException) => {
+                if (error.response && error.response.data && error.response.data.description) {
+                    dispatch(
+                        addAlert({
+                            description: t("authenticationProvider:notifications.getIDP.error.description", {
+                                description: error.response.data.description
+                            }),
+                            level: AlertLevels.ERROR,
+                            message: t("authenticationProvider:" + "notifications.getIDP.error.message")
+                        })
+                    );
+
+                    return;
+                }
+
+                dispatch(
+                    addAlert({
+                        description: t("authenticationProvider:" + "notifications.getIDP.genericError.description"),
+                        level: AlertLevels.ERROR,
+                        message: t("authenticationProvider:" + "notifications.getIDP.genericError.message")
+                    })
+                );
+            })
+            .finally(() => {
+                setConnectorDetailFetchRequestLoading(false);
+            });
+    };
+
+
+    const resolveConnectorImage = (connector: CustomAuthConnectionInterface): ReactElement => {
+        if (!connector) {
+            return;
+        }
+
+        return (
+            <AppAvatar
+                hoverable={ false }
+                name={ connector?.displayName }
+                image={ AuthenticatorMeta.getAuthenticatorIcon(
+                    connector?.image
+                ) }
+                size="tiny"
+            />
+        );
+    }
+
+    /**
+     * Handles the back button click event.
+     */
+    const handleBackButtonClick = (): void => {
+        history.push(AppConstants.getPaths().get("IDP"));
+    };
+
+    /**
+     * Called when an idp is deleted.
+     */
+    const handleIdentityProviderDelete = (): void => {
```

Comment:
The function name is misleading as it doesn't *handle* idp deletion. Shall we try following instead?

```suggestion
    const onIdentityProviderDelete = (): void => {
```